### DISCUSSION TITLE
Script exit code

### DISCUSSION THREAD

**Discussion by mvuidev** [polybar/polybar#2857] [Q&A] [answered]
Hello,

I have a simple script module which exec a .sh file and try to get label-fail working.

Basically :
```sh
#!/bin/bash

if [[ condition ]]; then
    echo "something"
else
    exit 1
fi
```
So if the condition fails, label-fail would be used.
But the module just disappears from the bar.

What's the right way to achieve this ?

**Comment by patrick96** [comment, marked as answer]
Can you show your module config?

**Reply by mvuidev** [reply]
Here it is (Fedora 36 / polybar 3.6.3) :

```ini
[module/vpn]
type = custom/script
interval = 5
tail = false

format = <label>
format-fail = <label-fail>
format-fail-background = ${color.shade12}

label = "VPN ON"
label-fail = "VPN OFF"

exec = ~/.config/polybar/scripts/vpn
```

So the label "VPN ON" should be displayed if my sh script echo something, and the label-fail "VPN OFF" should be displayed if the script exit 1.

But maybe I missed something.
Thank you for your help.

**Reply by patrick96** [reply]
Ah, I see what the problem is. The script module completely disappears if the script did not produce any output. But I don't think that behavior makes sense if the script actually failed (which is what happens in your case).

**Reply by patrick96** [reply]
Changed in #2861. This will be in the next release. For now, just make sure your script produces an output if it fails.

**Reply by mvuidev** [reply]
Nice !
For now I'll stick with an output to handle the fail.

Thank you, have a nice weekend.
